### PR TITLE
[explorer] All changes from gdc demo

### PR DESCRIPTION
**Pull request by stella3d**
All changes under the `./explorer` folder on the `explorer-rest` branch that were made in the run up to GDC.

It is a lot of changes, but picking them apart might be difficult.  

The changes to the explorer assume that we have some of the changes to the backend REST api from `experimental-rest-api` branch, so we want to get that merged in a similar timeframe ideally.

### CODE REVIEW

File: explorer/client/src/app/App.tsx [hunk 1]
```diff
@@ -1,8 +1,4 @@
-// Copyright (c) 2022, Mysten Labs, Inc.
```
Comment (666lcz) [inline explorer/client/src/app/App.tsx:1]: This is from a pr merged this morning. I think you might need to rebase?

File: explorer/client/conversion-scripts/move-json-converter.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const fs = require('fs');
```
Comment (666lcz) [inline explorer/client/conversion-scripts/move-json-converter.js:1]: Do we need to commit this folder `explorer/client/conversion-scripts`?

File: explorer/client/src/pages/address-result/AddressResult.tsx [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    const defaultData = (addressID: string | undefined) => ({ id: addressID, objects: [{}], loadState: 'pending' });
+    const [data, setData] = useState(defaultData(addressID));
+
+    useEffect(() => {
+        if(addressID === undefined)
+            return;
+
+        rpc.getAddressObjects(addressID)
+        .then((json) => {
+            console.log(json);
```
Comment (666lcz) [inline explorer/client/src/pages/address-result/AddressResult.tsx:35]: nit: remove

File: explorer/client/src/utils/utility_functions.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import mockTransactionData from './mock_data.json';
+import { DefaultRpcClient as rpc } from './rpc';
+
+
+const navigateWithUnknown = async (input: string, navigate: Function) => {
+    console.log('searching for ' + input);
```
Comment (666lcz) [inline explorer/client/src/utils/utility_functions.tsx:6]: nit: remove this and several other places in the file as well

File: explorer/client/src/utils/utility_functions.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import mockTransactionData from './mock_data.json';
```
Comment (666lcz) [inline explorer/client/src/utils/utility_functions.tsx:1]: Similar to Sam's [comment](https://github.com/MystenLabs/sui/pull/454#discussion_r807128575), let's avoid having a general util file and create `navigation_utils`, `bytes_conversion_utils`, `logging_util`.. instead, or move to the file in which they are used if there's only one

File: explorer/client/src/utils/rpc_basictest.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { SuiRpcClient } from "./rpc";
```
Comment (666lcz) [inline explorer/client/src/utils/rpc_basictest.ts:1]: Should we remove this or write better test?

File: explorer/client/.eslintrc.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
     extends: ['react-app', 'react-app/jest', 'prettier'],
     rules: {
-        'react/jsx-no-bind': ['error'],
+        'react/jsx-no-bind': ['warn'],
```
[the remark this answers is not in the document]
Comment (stella3d) [reply explorer/client/.eslintrc.js]: https://github.com/MystenLabs/sui/pull/1041/commits/59434a13e91eede8f8c4a0c5d03de97f918328d4

File: explorer/client/src/app/App.tsx (hunk 1 above)
Comment (stella3d) [reply explorer/client/src/app/App.tsx:1]: from main ? 
i checked a new branch out from main and brought over all the explorer changes

File: explorer/client/src/utils/rpc_basictest.ts (hunk 6 above)
Comment (stella3d) [reply explorer/client/src/utils/rpc_basictest.ts:1]: remove, i was merely using this to get the rpc lib up and running

Comment (stella3d) [reply explorer/client/src/utils/rpc_basictest.ts]: removed: https://github.com/MystenLabs/sui/commit/a8e02eaf8e18e9cf7e6b93d5f541bbde5210ab9f

File: explorer/client/src/app/App.tsx (hunk 1 above)
Comment (666lcz) [reply explorer/client/src/app/App.tsx:1]: Yes: https://github.com/MystenLabs/sui/pull/1028. It was merged to main 7 hours ago

File: explorer/client/src/utils/utility_functions.tsx (hunk 5 above)
Comment (stella3d) [reply explorer/client/src/utils/utility_functions.tsx:1]: ok, i'll  see what organization makes sense and move things around

File: explorer/client/src/pages/address-result/AddressResult.tsx (hunk 3 above)
Comment (stella3d) [reply explorer/client/src/pages/address-result/AddressResult.tsx]: removed: https://github.com/MystenLabs/sui/commit/b2816d50ea26fcbeded179d58dd2257efd9c0f10

File: explorer/client/src/utils/utility_functions.tsx (hunk 4 above)
Comment (stella3d) [reply explorer/client/src/utils/utility_functions.tsx]: removed: https://github.com/MystenLabs/sui/commit/3ec57a29a5863f26f997e70d658ef87d2c8f5a0f

File: explorer/client/conversion-scripts/move-json-converter.js (hunk 2 above)
Comment (stella3d) [reply explorer/client/conversion-scripts/move-json-converter.js:1]: that's a good question, i think @Andrew47 might know more about this than me

**Comment by Andrew47** [comment]
When `https://demo-rpc.sui.io/objects?address=` is applied to an object, the backend responds with `[]` rather than an error message. I've implemented a block that prevents address results from showing in this case.

If you visit https://explorer-demo.sui.io/objects/4ea58a9788562c53fe11f63e2662401d8312b14d and click the Owner value and then click back you will see the problem this is causing.

File: explorer/client/src/utils/utility_functions.tsx (hunk 5 above)
Comment (stella3d) [reply explorer/client/src/utils/utility_functions.tsx]: reorganized into 3 files:
`searchUtil.ts` : search function
`mockUtil.ts`: mock data function
`stringUtils.ts` : string manipulation & testing functions

**Comment by 666lcz** [comment]
Please run 

```
find ./ -name "*.ts" -exec gsed -i -e '1i\/\/ Copyright (c) 2022, Mysten Labs, Inc.\n\/\/ SPDX-License-Identifier: Apache-2.0\n' '{}' \;\n

find ./ -name "*.tsx" -exec gsed -i -e '1i\/\/ Copyright (c) 2022, Mysten Labs, Inc.\n\/\/ SPDX-License-Identifier: Apache-2.0\n' '{}' \;\n

```

to fix all the license headers, otherwise you will fail this [check](https://github.com/MystenLabs/sui/pull/1051).

**Review by 666lcz — APPROVED** [review]
Approve to unblock

**Comment by stella3d** [comment]
@666lcz that command doesn't seem to work, always errs with ```find: gsed: No such file or directory```

**Comment by 666lcz** [comment]
`brew install gsed`

**Comment by Andrew47** [comment]
I've created a PR here https://github.com/MystenLabs/sui/pull/1070. This fixes some bugs, reinstates a working UI test suite, fixes latency issues and splits the view/fetch logic. It also resolves the above issue with `Required statuses must pass before merging`. The branch has been rebased on this branch to remove any conflicts

**Review by Clay-Mysten — COMMENTED** [review]
Retroactive thanks to Stella for excellent doc additions.

**Pull request merged**